### ISSUE TITLE
LG Air fan speed control with Homekit bug 

### ISSUE THREAD

**Issue by crackn** [hristo-atanasov/Tasmota-IRHVAC#101]
hello, 
is there a way to change fan speed mode from "min" to "low" in LG config because Apple homekit only accept low/medium/high and LG air accept min/medium/high or min/medium/max if i just change the min mode to low, LG refuses the command. 

so to tasmota IR the command must be FanSpeed: "min" but for homeassistant the command must be "Low" 

thanks,

**Comment by hristo-atanasov** [comment]
Currently I can't imagine a way to make this to work. I can only propose a way how you can change the code (locally on your machine) to make it work for you. But this mean, that you'll have a custom version of the custom component. This means, that every time there is a new version, you will have to update and change the code on few places.  :)

**Comment by crackn** [comment]
I see no problem with it. If you
can elaborate how to do it, I will make the changes and test it/report.

Thanks.

Enviado do meu iPhone

**Comment by hristo-atanasov** [comment]
I made the changes you need in separate branch. Try this branch and tell me if it works for you:
https://github.com/hristo-atanasov/Tasmota-IRHVAC/tree/for-crackn
There are only two changes marked witch "# crackn" so it would be easier for you to find them. If it is working, you can fork it, but remember, that you have to apply this changes each time there is a new version. :)

**Comment by crackn** [comment]
going to try it right now, I will let you know if it works. 

thanks

**Comment by crackn** [comment]
Yes it works. Now I can control it from homeassistant interface and from Homekit using apple Home app. 

now I can say that integration with LG is working 100%

thanks alot for your help.

**Comment by crackn** [comment]
hello, could you verify what changed in new versions about this bug, It's not working and I cant merge the code you did for LG problem. thanks in advance.

**Comment by crackn** [comment]
i tried to merge the new branch with the one modified for my user. I can't. I dont know how to make it. would you mind update my branch with  the latest version? thanks!

**Comment by hristo-atanasov** [comment]
Sure! I'll have a look when I have some time.

**Comment by crackn** [comment]
ok, thanks!

**Comment by hristo-atanasov** [comment]
> ok, thanks!

Can you check if it works for you again?

**Comment by crackn** [comment]
yes it works, one thing I noted but i think its an old bug, is that when o set the horizontal swing to ON, it stops, when I set it to OFF, it swings, is there a way to just invert the commands? 

cheers,
cRACKn

**Comment by hristo-atanasov** [comment]
> yes it works.
> 
> btw, one thing I noted but i think its an old bug, is that when I set the horizontal swing to ON, it stops, when I set it to OFF, it swings, is there a way to just invert the commands?
> 
> cheers, cRACKn

I'll have a look at it .. :)

**Comment by crackn** [comment]
hello, 
Could you fix the patch once again, I cannot merge the main code commits with the version you did last time.

thanks in advance.

**Comment by nemiro-TECH** [comment]
It would be easier to expose your HA entities to your Homekit...
